core/src/main/java/hudson/Util.java

```diff
@@ -1593,7 +1593,7 @@ public static long daysElapsedSince(@Nonnull Date date){
      * give up, thus improving build reliability.
      */
     @Restricted(value = NoExternalUse.class)
-    static int DELETION_MAX = Math.max(1, SystemProperties.getInteger(Util.class.getName() + ".maxFileDeletionRetries", 3));
+    static int DELETION_MAX = Math.max(0, SystemProperties.getInteger(Util.class.getName() + ".maxFileDeletionRetries", 2));
```

Comment:
keep the same behaviour as before (we will try and delete the file 3 times).  
this will change the behaviour if someone had specified a number - but then we actually will now do what they asked us to so I consider that appropriate

Reply:
NIT: It might be more appropriate to change the name of the variable as well `DELETION_RETRIES` perhaps. In order to ensure that future readers are not confused.